docs/design/2022-08-29-sync-stats-for-optimizer.md

```diff
@@ -0,0 +1,73 @@
+
+For example, for the query `select a from t where b = 1 and c = 1` and execution its plan is like following:
+
+```sql 
++---------------------------+----------+-----------+---------------+----------------------------------+
+| id                        | estRows  | task      | access object | operator info                    |
++---------------------------+----------+-----------+---------------+----------------------------------+
+| Projection_4              | 0.01     | root      |               | test.t.a                         |
+| └─TableReader_7           | 0.01     | root      |               | data:Selection_6                 |
+|   └─Selection_6           | 0.01     | cop[tikv] |               | eq(test.t.b, 1), eq(test.t.c, 1) |
+|     └─TableFullScan_5     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo   |
++---------------------------+----------+-----------+---------------+----------------------------------+
+```
+
+The column `b` and `c` are maintained in the `Selection_6` and `collectPredicateColumnsPoint` will extract them in the operator so that we can know what's the required stats.
+After collecting the columns and indices from the queries, we will also check them whether they are necessary to retrieve the stats.
+For each column and index maintained in the tidb-server, tidb-server will also maintain the stats loading status in memory as `StatsLoadedStatus`.
+If the `StatsLoadedStatus` of column or index is full loaded, it will not to load stats during logical optimizing.
+
+#### Retrieve the stats
+
+After collecting the necessary columns and indices, tidb-server will collect the stats during `syncWaitStatsLoadPoint` phase.
+For each tidb-server, it will start several sub sync stats loader to load stats for columns and indices given by the above phase.
+As each query may send loading stats request to sync stats loader, the required columns and indices might be duplicated between each query.
+Thus in order to improve the performance for sync stats loader and reduce loading duplicated stats for columns and indices, each sync stats loader will check other loaders in order to avoid loading duplicated stats.
+If the required stats are loaded by one loader, other loaders will not load this stats again.
+
+## Diagnosability
+
+If any error happened during `Sync Stats`, tidb-server should provide the ability to diagnose the error happened in the sync stats.
+
+### Slow Log
+
+If any error happened during sync stats, it will set up a flag in slow log as `IsSyncStatsFailed` to indicate the there existed error during sync stats for this query.
+
+And the stats info will also be recorded in the slow log including its loading memory status, this will also be helpful for sync stats loader whether it load stats successfully.
```

Comment:
```suggestion
And the stats info will also be recorded in the slow log including its loading memory status, this will also be helpful to check whether stats is loaded successfully.
```